src/raftstore/store/snap.rs

```diff
@@ -509,7 +509,7 @@ mod v1 {
                     let cap = cmp::min(self.left, DEFAULT_READ_BUFFER_SIZE);
                     let mut buf = vec![0; cap];
                     while self.left > 0 {
-                        try!(self.read(&mut buf));
+                        try!(self.read_exact(&mut buf));
```

Comment:
If not usning read_exact, then we must handle read amount returned. Or it'll cause a clippy error.

error: handle read amount returned or use `Read::read_exact` instead
   --> src/raftstore/store/snap.rs:512:25
    |
512 |                         try!(self.read(&mut buf));
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: #[deny(unused_io_amount)] on by default
    = help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#unused_io_amount
    = note: this error originates in a macro outside of the current crate

error: aborting due to previous error

error: Could not compile `tikv`.